woocommerce/woocommerce
# Display the coming soon page on front-end requests

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Displays the coming soon page on applicable front-end requests when the site is in coming soon mode. Depending on the coming soon settings, either the whole site or just the store pages will be protected. It displays the coming soon page defined in the advanced settings.

Logged in admins do not see the coming soon page -- instead they will see a footer banner handled outside this PR.

Uses the [container](https://github.com/woocommerce/woocommerce/blob/c2f486b407883e6e7462ac5f8402647375c927d3/plugins/woocommerce/src/README.md#the-container) and dependency injection to load the classes.


Closes #46222  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the WooCommerce Admin feature flag `launch-your-store`
2. Go to WooCommerce settings and set the site to coming soon mode.
3. As an admin user, see that the front-end is shown normally (or with a footer banner).
4. Visit the site home screen from a new private window.
5. See that the coming soon page is displayed.
6. In the WooCommerce settings, restrict coming soon to store pages only.
7. Reload the homescreen in the private window and see that the coming soon page is **not** displayed.
8. Navigate to a store page like the cart.
9. See that the coming soon page is displayed



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
